scalanet/test/src/io/iohk/scalanet/peergroup/kademlia/Generators.scala

```diff
@@ -65,12 +74,15 @@ object Generators {
     BitVector.bits(Range(0, bitLength).map(_ => Random.nextBoolean()))
 
   def aRandomNodeRecord(
-      bitLength: Int = defaultBitLength
+      bitLength: Int = defaultBitLength,keyPair:Option[(ECPrivateKeyParameters,ECPublicKeyParameters)] = None
   ): NodeRecord[String] = {
-    NodeRecord(
-      id = aRandomBitVector(bitLength),
+    val pair = if(keyPair.isEmpty) crypto.generateKeyPair(random) else keyPair.get
```

Comment:
val (privateKey, publicKey) = if ...

btw, instead of using Option, could we just set default value `keyPair = crypto.generateKeyPair(random)` ?